packages/block-editor-tools/src/hooks/use-post-by-id/README.md

```diff
@@ -0,0 +1,35 @@
+# Custom Hooks: usePostById
+
+A custom React hook to retrieve post data given only a post ID.
+If you have the post type, use `usePost` instead.
+
+## Usage
+
+```jsx
+const MyBlock = ({
+	postID,
+}) => {
+  const post = usePostById(postID);
+
+  if (post) {
+    ...
+  }
+};
+```
+
+You can also pass a function to lookup the post type when passed the post id.
```

Comment:
Note that whatever gets returned by the custom function needs to return an object with similar structure to what gets returned by the search endpoint, and at an absolute minimum needs a `subtype` key with the post type slug in it.